creator-node/src/diskManager.ts

```diff
@@ -565,6 +570,7 @@ async function _migrateDirsWithLegacyStoragePaths(
     let prevLegacyStoragePathsAndCids
     let legacyStoragePathsAndCids: { storagePath: string; cid: string }[] = []
     while (!isEqual(prevLegacyStoragePathsAndCids, legacyStoragePathsAndCids)) {
```

Comment:
I'll think about how this can be made clearer. your code block makes sense to me except in the edge case where the query keeps returning the same files that fail to copy so then they don't get updated in db

Reply:
how does this look?
```ts
let newResultsFound = true
let results = []
while (newResultsFound) {
  const prevResults = results
  results = await DbManager.getDirLegacyStoragePathsAndCids(
    cursor,
    BATCH_SIZE
  )
  if (results.length) {
    newResultsFound = !isEqual(prevResults, results)
    ...
  } else {
    newResultsFound = false
    ...
  }
}
```